src/ol_infrastructure/components/services/k8s.py

```diff
@@ -0,0 +1,311 @@
+
+
+stack_info = parse_stack()
+env_name = f"{stack_info.env_prefix}-{stack_info.env_suffix}"
+
+
+class OLApplicationK8s(ComponentResource):
+    """
+    Main K8s component resource class
+    """
+
+    def __init__(
+        self,
+        ol_app_k8s_config: OLApplicationK8sConfiguration,
+        opts: Optional[ResourceOptions] = None,
+    ):
+        """
+        It's .. the constructor. Shaddap Ruff :)
+        """
+        super().__init__(
+            "ol:infrastructure:components:services:OLApplicationK8s",
+            ol_app_k8s_config.application_namespace,
+            None,
+            opts=opts,
+        )
+        resource_options = ResourceOptions(parent=self).merge(opts)
+        self.ol_app_k8s_config: OLApplicationK8sConfiguration = ol_app_k8s_config
+
+        if ol_app_k8s_config.import_nginx_config:
+            application_nginx_configmap = kubernetes.core.v1.ConfigMap(
+                f"unified-application-{stack_info.env_suffix}-nginx-configmap",
+                metadata=kubernetes.meta.v1.ObjectMetaArgs(
+                    name="nginx-config",
+                    namespace=ol_app_k8s_config.application_namespace,
+                    labels=ol_app_k8s_config.k8s_global_labels,
+                ),
+                data={
+                    "web.conf": ol_app_k8s_config.project_root.joinpath(
+                        "files/web.conf"
+                    ).read_text(),
+                },
+                opts=resource_options,
+            )
+
+        # Build a list of not-sensitive env vars for the deployment config
+        application_deployment_env_vars = []
+        for k, v in (self.ol_app_k8s_config.application_config).items():
+            application_deployment_env_vars.append(
+                kubernetes.core.v1.EnvVarArgs(
+                    name=k,
+                    value=v,
+                )
+            )
+        application_deployment_env_vars.append(
+            kubernetes.core.v1.EnvVarArgs(name="PORT", value=str(DEFAULT_UWSGI_PORT))
+        )
+
+        # Build a list of sensitive env vars for the deployment config via envFrom
+        application_deployment_envfrom = [
+            # Database creds
+            kubernetes.core.v1.EnvFromSourceArgs(
+                secret_ref=kubernetes.core.v1.SecretEnvSourceArgs(
+                    name=self.ol_app_k8s_config.db_creds_secret_name,
+                ),
+            ),
+            # Redis Configuration
+            kubernetes.core.v1.EnvFromSourceArgs(
+                secret_ref=kubernetes.core.v1.SecretEnvSourceArgs(
+                    name=self.ol_app_k8s_config.redis_creds_secret_name,
+                ),
+            ),
+            # static secrets from secrets-application/secrets
+            kubernetes.core.v1.EnvFromSourceArgs(
+                secret_ref=kubernetes.core.v1.SecretEnvSourceArgs(
+                    name=self.ol_app_k8s_config.static_secrets_name,
+                ),
+            ),
```

Comment:
It's _possible_ that we might have applications that only require a subset (or superset) of secrets objects. Maybe consider rolling the secret names into an iterable object that can be processed in this context.

Reply:
I feel like the iterable approach would be an excellent refinement. Let's go with this for simplicity for now.